packages/generator-rest/src/generator.ts

```diff
@@ -13,71 +13,128 @@ import {
+    generateSourcesForService(metadata, project, options)
+  );
+  await project.save();
+}
 
-      logger.info(`Argument for openapi generator ${generationArguments}`);
-
-      try {
-        const response = await execa('npx', generationArguments).catch(err => {
-          logger.error('In exception block 2');
-          logger.error(err);
-        });
-        if (response !== undefined) {
-          logger.info(`Generated the client ${response.stdout}`);
-        }
-      } catch (err) {
-        logger.error('In exception block');
-        logger.error(err);
-      }
-    })
+async function generateOneApi(
+  file: string,
+  options: GeneratorOptions,
+  pathToTemplates: string,
+  pathToMustacheValues: string
+) {
+  const folderForService = getFolderForService(options.outputDir, file);
+  mkdirSync(folderForService);
+
+  const serviceName = getServiceNameCamelCase(file);
+  const adjustedOpenApiContent = getAdjustedOpenApiFile(
+    join(options.inputDir, file),
+    serviceName
+  );
+  const pathToAdjustedOpenApiDefFile = join(folderForService, 'open-api.json');
+  writeJsonSync(pathToAdjustedOpenApiDefFile, adjustedOpenApiContent);
+  // todo
+  await generateFilesUsingOpenAPI(
+    folderForService,
+    pathToAdjustedOpenApiDefFile,
+    pathToTemplates,
+    pathToMustacheValues
   );
+
+  const openApiModel = toOpenApiModel(
+    pathToAdjustedOpenApiDefFile,
+    serviceName,
+    folderForService
+  );
+  return openApiModel;
+}
+
+function generateSourcesForService(
+  serviceMetadata: OpenApiServiceMetadata,
+  project: Project,
+  options: GeneratorOptions
+) {
+  const serviceDir = project.createDirectory(
+    resolve(options.outputDir.toString(), serviceMetadata.serviceDir)
+  );
+  logger.info(`Generating request builder in ${serviceDir}.`);
+  sourceFile(
+    serviceDir,
+    'request-builder',
+    requestBuilderSourceFile(serviceMetadata),
+    true
+  );
+}
+
+async function generateFilesUsingOpenAPI(
+  folderForService: string,
+  pathToAdjustedOpenApiDefFile: string,
+  pathToTemplates: string,
+  pathToMustacheValues: string
+) {
+  const generationArguments = [
+    'openapi-generator',
+    'generate',
+    '-i',
+    pathToAdjustedOpenApiDefFile,
+    '-g',
+    'typescript-axios',
+    '-o',
+    folderForService,
+    '-t',
+    pathToTemplates,
+    '--api-package',
+    'api',
+    '--model-package',
+    'model',
+    '--config',
+    pathToMustacheValues,
+    '--skip-validate-spec'
+  ];
+
+  logger.info(`Argument for openapi generator ${generationArguments}`);
+
+  try {
+    const response = await execa('npx', generationArguments).catch(err => {
+      logger.error('In exception block 2');
+      logger.error(err);
+    });
+    if (response !== undefined) {
+      logger.info(`Generated the client ${response.stdout}`);
+    }
+  } catch (err) {
+    logger.error('In exception block');
+    logger.error(err);
+  }
 }
 
 function getServiceNameCamelCase(openApiFileName: string) {
```

Comment:
I know this has been here before, but maybe this can be put in a separate module (like a name formatter). Also `getServiceNameCamelCase` calls `toPascalCase`.

Reply:
Will move to core

Reply:
This is really open-api/json related implementation. I guess this should not be reused outside the package.